src/ess/dream/beamline.py

```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+"""Beamline tools for DREAM."""
+
+from ess.powder.types import DetectorDimensions, DetectorName
+
+
+def dream_detector_dimensions(detector_name: DetectorName) -> DetectorDimensions:
+    """Logical dimensions used by a DREAM detector.
+
+    Parameters
+    ----------
+    detector_name:
+        Name of a detector of DREAM.
+
+    Returns
+    -------
+    :
+        Logical dimensions used by the given DREAM detector.
+    """
+    base = ('module', 'segment', 'counter', 'wire', 'strip')
+    if detector_name == DetectorName.high_resolution:
+        return DetectorDimensions(base + ('sector',))
```

Comment:
I have not encountered this. But I only have simulated data. Do you have a NeXus file where it doesn't work? If so, we need to update the csv loader.

Reply:
See https://github.com/scipp/essdiffraction/blob/main/src/ess/dream/io/nexus.py.

Reply:
>  The endcap detectors have a irregular structure that cannot be fully folded.
      This is not a problem but note again that the counter may be reversed. It is
      currently not clear if this is a bug in the file.

What do I do with this information? Do I just not fold the endcaps? Or do I only fold part of them, which part?
Or Do we just wait because it seems to be unclear whether real files will be like this?

Reply:
I am fine with any discrepancy, providing that this code is ensured to be Geant-4 specific.